src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -103,31 +104,38 @@ public async IAsyncEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery q
             int tableCompletionCount = 0;
 
             using IEnumerator<ProgressiveDataSetFrame> frames = pDataSet.GetFrames();
-            
+
             while (frames.MoveNext())
             {
                 var frame = frames.Current;
 
                 switch (frame.FrameType)
                 {
                     case FrameType.TableFragment:
+                    {
+                        var content = frame as ProgressiveDataSetDataTableFragmentFrame;
+                        while (GetNextRow(content, out object[] row))
                         {
-                            var content = frame as ProgressiveDataSetDataTableFragmentFrame;
-                            while(GetNextRow(content, out object[] row))
-                            {
-                                yield return row;
-                            };
+                            yield return row;
                         }
+                    }
                         break;
 
                     case FrameType.DataTable:
+                    {
+                        // Note from documentation: we can't skip processing the data -- we must consume it.
+                        var content = frame as ProgressiveDataSetDataTableFrame;
+                        var reader = content.TableData;
+                        while (reader.Read())
                         {
-                            var content = frame as DataTable;
-                            foreach (DataRow dataTableRow in content.Rows)
-                            {
-                                yield return dataTableRow.ItemArray;
-                            }
+                            var writer = new System.IO.StringWriter();
+                            reader.WriteAsText("", true, writer,
```

Comment:
I don't think we could return anything from this FrameType, or what do we want to return from the  ProgressiveDataSetDataTableFrame? 
on the overall code we are returning objects but, on this case, we don't have any object to return, and it doesn't' have any record on it to get objects. 

The ProgressiveDataSetDataTableFrame only have the TableData data is a IDataReader (and in the documentation says that we need to consume) when we don't consume it, we get the error:
` System.InvalidOperationException : Cannot move to the next frame while the current frame ('DataTable') is not fully materialized.`

Also, when we consume it all the information that has in it is: 
```
 Result Set:
        -----------------------------
        TableId  | Key     | Value   
        int      | string  | dynamic 
        -----------------------------
```
 I don't think this needs to be returned, or are we expecting something different from this FrameType?

Reply:
Maybe this isn't the correct frame type we want, then. (Which is part of the problem with the lack of documentation from this team). If I had more time to work on this, I would've tried to run our query that we were going to use this functionality for through this code to see what types it would return, and then refactor the code depending on the types we were getting. If you're able to build out a quick test to run locally to see what it will return, at least that will help us narrow down which frame types are the ones we want to care about. I'm pretty sure that the "Progressive" frame types are the ones that are going to give us the "streaming" data, but Kusto might not even return that to us at all.

Reply:
I ran some of the queries that we are going to run using this, the information is on FrameType.TableFragment and we are returning that part correctly:

```
 var content = frame as ProgressiveDataSetDataTableFragmentFrame;
while (GetNextRow(content, out object[] row))
{
    yield return row;
}
```

on this part we are actually returning the rows of the table. 

I ran the query with the code as it's on this pr, I ran it with a small query and with a big one (one with the Kusto limits errors) and in both cases all the rows were returned as we expected.